### DISCUSSION TITLE
17 gennaio 2024 B - esercizio 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#93] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/17d014f4-ed59-4e6f-a9fb-9968d5f64e01)

**Comment by daromacarlo** [comment]
![Immagine 2024-02-05 003122](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/446f20e2-a5a8-40f9-a98e-de6b930638d1)
![Immagine 2024-02-05 003136](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/5d5e2396-319a-49da-9cb9-cf414a53be26)
![Immagine 2024-02-05 003148](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/396c71f2-1239-49f6-a115-4dfc7c1ab25f)
![Immagine 2024-02-05 003200](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/fccb3df3-0e8b-4615-89df-8bb3193d62ae)
![Immagine 2024-02-05 003220](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/ff8f5334-6292-4db8-8980-09e078957152)

**Reply by Frisbon** [reply]
Penso ci sia un piccolo errore all'inizio

$AH^+ =$
$ABDEH \text{ perché } (A\rightarrow D \wedge H \rightarrow E \wedge HA \rightarrow B) = $
$ABCDEH \text{ perché } (BD \rightarrow CH) = R \implies K$

Inoltre, nell'algoritmo di decomposizione, dovrebbe esserci un OR (Se gli appunti di Exyss non mentono) nel primo if

**Reply by daromacarlo** [reply]
grazie, ricontrollerò

**Reply by daromacarlo** [reply]
@Frisbon ho ricontrollato, hai ragione.

**Reply by salicius** [reply]
Nell'esercizio 3b che sarebbe il 2c, nel trovare la copertura minimale andava splittato anche HA->B giusto ? Inoltre puoi dirmi che significa che hai barrato quelle dipendenze ? (quella con croce verde e viola)

**Reply by daromacarlo** [reply]
@salicius non ricordo bene sinceramente🤔🤔🤔, ma quando barro significa che sono chiavi ridondanti.

**Comment by Frisbon** [comment]
### Esercizio strano, qualcuno puó verificarne la validitá?

![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/84731341/80db482d-dbea-4f0c-bed1-f4b8226284d4)

**Reply by niccolo-fato** [reply]
a me la decomposizione viene {BDA, BDC, AD, CE, CDH, HE ,HAB}

**Reply by IonutCicio** [reply]
Potrebbe succedere che vengano fuori due decomposizioni diverse, perché nel calcolo della copertura minimale magari uno leva una dipendenza piuttosto che un'altra, e alla fine la decomposizione viene diversa. Detto questo, non ho verificato se il suo procedimento è giusto o meno.

**Reply by Frisbon** [reply]
> a me la decomposizione viene {BDA, BDC, AD, CE, CDH, HE ,HAB}

Probabilmente é giusta la tua, ai tempi non sapevo come fare per bene l'esercizio.
